rmw_opendds_cpp/src/rmw_publisher.cpp

```diff
@@ -212,6 +213,10 @@ rmw_create_publisher(
       throw std::string("failed to create datawriter");
     }
 
+    OpenDDS::DCPS::DataWriterImpl_T<OpenDDSStaticSerializedData> * trt = dynamic_cast<OpenDDS::DCPS::DataWriterImpl_T<OpenDDSStaticSerializedData>*>(publisher_info->topic_writer_.in());
+
+    trt->set_marshal_skip_serialize(true);
```

Comment:
Wouldn't this be a candidate for a QoS extension in order to not use internal OpenDDS classes?

Reply:
Could you explain what you mean a little more? I like the idea of not using OpenDDS namespace items in the RMW but we have about a dozen source files doing so at the moment. In other words, I'm not sure how important that is as a design goal in the RMW.

Reply:
Could this be done with an extension of one of the QoS structs so that you can specify a custom QoS in your code to enable this internal feature? Not sure how generic this all can be used.

Reply:
This should not be used anywhere.  Even in RMW, it's only there as a work around.